$NetBSD: patch-CMakeLists.txt,v 1.3 2025/02/15 09:38:47 mef Exp $

disable finding the commit hash: we provide it in the Makefile

use pkgsrc mimalloc directly

--- CMakeLists.txt.orig	2022-05-18 12:09:48.000000000 +0900
+++ CMakeLists.txt	2025-02-15 17:37:23.398344272 +0900
@@ -34,7 +34,6 @@ endif()
 # NOTE TO PACKAGERS: The embedded git commit hash is critical for rapid bug triage when the builds
 # can come from a variety of sources. If you are mirroring the sources or otherwise build when
 # the .git directory is not present, please comment the following line:
-include(GetGitCommitHash)
 # and instead uncomment the following, adding the complete git hash of the checkout you are using:
 # set(GIT_COMMIT_HASH 0000000000000000000000000000000000000000)
 
@@ -185,14 +184,6 @@ endif()
 message(STATUS "Using in-tree libdxfrw")
 add_subdirectory(extlib/libdxfrw)
 
-message(STATUS "Using in-tree mimalloc")
-set(MI_OVERRIDE OFF CACHE BOOL "")
-set(MI_BUILD_SHARED OFF CACHE BOOL "")
-set(MI_BUILD_OBJECT OFF CACHE BOOL "")
-set(MI_BUILD_TESTS OFF CACHE BOOL "")
-add_subdirectory(extlib/mimalloc EXCLUDE_FROM_ALL)
-set(MIMALLOC_INCLUDE_DIR ${CMAKE_SOURCE_DIR}/extlib/mimalloc/include)
-
 if(NOT FORCE_VENDORED_Eigen3)
     find_package(Eigen3 CONFIG)
 endif()
@@ -269,6 +260,7 @@ else()
     find_package(ZLIB REQUIRED)
     find_package(PNG REQUIRED)
     find_package(Freetype REQUIRED)
+    find_package(mimalloc REQUIRED)
     pkg_check_modules(CAIRO REQUIRED cairo)
 endif()
 
